src/vendor/command/l2cap.rs

```diff
@@ -361,20 +349,25 @@ pub struct L2CapCocReconfig {
     /// - 0 .. 5
     pub channel_number: u8,
     /// List of channel indexes for which the primitives apply.
-    pub channel_index_list: [u8; 246],
+    pub channel_index_list: [u8; 5],
 }
 
 impl L2CapCocReconfig {
-    const MAX_LENGTH: usize = 254;
+    const MIN_LENGTH: usize = 7;
+    const MAX_LENGTH: usize = 12;
 
     fn copy_into_slice(&self, bytes: &mut [u8]) {
-        assert!(bytes.len() >= Self::MAX_LENGTH);
+        assert!(bytes.len() >= Self::MIN_LENGTH);
+        assert!(bytes.len() <= Self::MAX_LENGTH);
 
         LittleEndian::write_u16(&mut bytes[0..], self.conn_handle.0);
         LittleEndian::write_u16(&mut bytes[2..], self.mtu);
         LittleEndian::write_u16(&mut bytes[4..], self.mps);
         bytes[6] = self.channel_number;
-        bytes[7..].copy_from_slice(&self.channel_index_list);
+
+        if self.channel_number > 0 {
+            bytes[7..].copy_from_slice(&self.channel_index_list[..self.channel_number as usize]);
```

Comment:
There's no explicit check if direct indexing `channel_index_list[..channel_number]` is valid. Probably this is implied somewhere else?